scripts/up.sh

```diff
@@ -44,15 +65,23 @@ coredns_image="multicluster/coredns:latest"
 coredns_path="/tmp/coredns-${coredns_version}"
 if [ ! -d "${coredns_path}" ]; then
   pushd /tmp
-  curl -fLO "https://github.com/coredns/coredns/archive/refs/tags/v${coredns_version}.tar.gz"
-  tar xf v${coredns_version}.tar.gz
+  # Since the coredns make command uses the git command, we clone it with git
+  git clone --depth 1 https://github.com/coredns/coredns.git --branch v${coredns_version} --single-branch "${coredns_path}"
+
+  # coredns binary is not multi-arch, so we need to download the binary for the current platform
+  platform="$(detect_platform)"
+  coredns_binary_url="https://github.com/coredns/coredns/releases/download/v${coredns_version}/coredns_${coredns_version}_${platform}.tgz"
+  curl -fLO "${coredns_binary_url}"
+  tar xzf "coredns_${coredns_version}_${platform}.tgz"
+  chmod +x coredns
   popd
 fi
 pushd "${coredns_path}"
 if ! grep -q -F 'multicluster:github.com/coredns/multicluster' "plugin.cfg"; then
   sed -i -e 's/^kubernetes:kubernetes$/&\nmulticluster:github.com\/coredns\/multicluster/' "plugin.cfg"
 fi
 make
+cp ../coredns ./coredns
```

Comment:
Ditto, we have to rebuild the coredns binary here

Reply:
https://github.com/kubernetes-sigs/mcs-api/pull/94#discussion_r1904575177

🙇